ResearchKit/Common/ORKScaleSlider.m

```diff
@@ -60,17 +63,78 @@ - (id)initWithFrame:(CGRect)frame
         self.showThumb = NO;
         
         _axLastOutputTime = 0;
+        _thumbImageNeedsTransformUpdate = NO;
     }
     return self;
 }
 
-- (void)setShowThumb:(BOOL)showThumb
-{
+- (void)setShowThumb:(BOOL)showThumb {
     _showThumb = showThumb;
 }
 
-- (void)sliderTouched:(UIGestureRecognizer *)gesture
-{
+- (void)setVertical:(BOOL)vertical {
+    if (vertical != _vertical) {
+        _vertical = vertical;
+        self.transform = _vertical ? CGAffineTransformMakeRotation(-M_PI_2) : CGAffineTransformIdentity;
+        _thumbImageNeedsTransformUpdate = YES;
+    }
+}
+
+- (UIView *)thumbImageSubview {
+    UIView *thumbImageSubview = nil;
+    CGRect trackRect = [self trackRectForBounds:self.bounds];
+    CGRect thumbRect = [self thumbRectForBounds:self.bounds trackRect:trackRect value:self.value];
+    for (UIView *subview in self.subviews) {
+        if (CGRectEqualToRect(thumbRect, subview.frame)) {
+            thumbImageSubview = subview;
```

Comment:
I agree that it's hackish and not very future proof, but in my opinion the usage of the actual `thumbRect` for identifying the subview makes it quite safe. Doesn't it?

Anyway, yeah, using an explicit thumb would be better, but I didn't wan't to bother you guys for a thumb design. I'll add a comment.